### PR TITLE
add_constraints should raise error on address columns

### LINKED ISSUE #1594: Update_transformers should raise error on address columns
### Problem Description

_As a user, I want address columns to be processed correctly. Anything that could get in the way of that should be prevented._

### Expected behavior

- If a user tries to call `update_transformers` on a column name that is part of an address, raise the following error

```python
synthesizer.update_transformers(column_name_to_transformer={
    'zip_code': AnonymizedFaker()
})
```
```bash
Error: The 'zip_code' column is part of an address. You cannot modify transformers that are part of an address group.
```
### Additional context

- To do this issue, you will need to know what columns are part of an address. That should be done in #1593

### LINKED ISSUE #1595: add_constraints should raise error on address columns
### Problem Description

_As a user, I want address columns to be processed correctly. Anything that could get in the way of that should be prevented._

### Expected behavior

- If a user tries to call `add_constraints` on a column name that is part of an address, raise the following error

```python
my_constraint = {
    'constraint_class': 'FixedCombinations',
    'constraint_parameters': {
        'column_names': [ 'sex', 'zip_code']
    }
}

synthesizer.add_constraints(constraints=[my_constraint])
```
```bash
Error: The 'zip_code' column is part of an address. You cannot add constraints to columns that are part of an address group.
```
### Additional context

- To do this issue, you will need to know what columns are part of an address. That should be done in #1593

### PR DESCRIPTION
**Pull request by R-Palazzo**
Resolve #1595 
Resolve #1594

### CODE REVIEW

File: sdv/data_processing/data_processor.py [hunk 1]
```diff
@@ -290,6 +311,21 @@ def _validate_constraint_dict(self, constraint_dict):
...
+        else:
+            column_names = constraint_parameters.get('column_names')
+
+        columns_in_address = self._get_columns_in_address_transformer()
+        if columns_in_address and column_names:
+            address_constraint_columns = set(column_names) & set(columns_in_address)
+            if address_constraint_columns:
+                to_print = "', '".join(address_constraint_columns)
+                raise InvalidConstraintsError(
+                    f"The '{to_print}' column is part of an address. You cannot add constraints "
```
Comment (amontanez24) [inline sdv/data_processing/data_processor.py:325]: should we just assume plural here? ie. "The '{to_print}' columns are ..."

File: sdv/data_processing/data_processor.py [hunk 2]
```diff
@@ -124,6 +124,27 @@ def _check_import_address_transformers(self):
...
+            self._check_import_address_transformers()
+            return [
+                item for col_tuple, transformer in self.grouped_columns_to_transformers.items()
+                if isinstance(
+                    transformer, (
+                        rdt.transformers.RandomLocationGenerator,
+                        rdt.transformers.RegionalAnonymizer
+                    )
+                ) for item in col_tuple
+            ]
```
Comment (amontanez24) [inline sdv/data_processing/data_processor.py:144]: this might be easier to read in a normal for loop

**Review by amontanez24 — APPROVED** [review]
Mostly minor comments but I think this is good!

File: sdv/data_processing/data_processor.py [hunk 3]
```diff
@@ -103,6 +104,78 @@ def __init__(self, metadata, enforce_rounding=True, enforce_min_max_values=True,
...
+        Args:
+            anonymization_level (str):
+                The anonymization level for the address transformer.
+        """
+        locales = self._locales if self._locales else ['en_US']
+        self._check_import_address_transformers()
+        if anonymization_level == 'full':
+            return rdt.transformers.RandomLocationGenerator(locales=locales)
+        elif anonymization_level == 'street_address':
+            return rdt.transformers.RegionalAnonymizer(locales=locales)
```
Comment (pvk-developer) [inline sdv/data_processing/data_processor.py:160]: Can remove the `elif` and just `return rdt.transformers.RegionalAnonymizer(locales=locales)` or raise an error if none of the options

File: sdv/data_processing/data_processor.py [hunk 4]
```diff
@@ -408,15 +496,24 @@ def create_regex_generator(self, column_name, sdtype, column_metadata, is_numeri
         return transformer
 
     def _get_transformer_instance(self, sdtype, column_metadata):
+        transformer = self._transformers_by_sdtype[sdtype]
+        if transformer.__class__.__name__ == 'AnonymizedFaker':
```
Comment (pvk-developer) [inline sdv/data_processing/data_processor.py:500]: We have `AnonymizedFaker` imported, so we can just check if its instance of it.
`if isinstance(transformer, AnonymizedFaker)`

File: sdv/data_processing/data_processor.py [hunk 5]
```diff
@@ -408,15 +496,24 @@ def create_regex_generator(self, column_name, sdtype, column_metadata, is_numeri
...
+            is_lexify = transformer.function_name == 'lexify'
+            is_baseprovider = transformer.provider_name == 'BaseProvider'
+            if is_lexify and is_baseprovider:  # Defaults settings
```
Comment (pvk-developer) [inline sdv/data_processing/data_processor.py:503]: `Default settings`

**Review by pvk-developer — APPROVED** [review]
Some minor comments but other than that looks good :shipit:

**Pull request merged**